### PR TITLE
8331866: Add warnings for locale data dependence

### PR DESCRIPTION
**Pull request by naotoj**
In order to enlighten users to not depend on a particular version of the locale data, putting a note into the JDK vs. CLDR version chart would be appropriate.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change requires CSR request [JDK-8331965](https://bugs.openjdk.org/browse/JDK-8331965) to be approved

### Issues
 * [JDK-8331866](https://bugs.openjdk.org/browse/JDK-8331866): Add warnings for locale data dependence (**Bug** - P4)
 * [JDK-8331965](https://bugs.openjdk.org/browse/JDK-8331965): Add warnings for locale data dependence (**CSR**)

### Reviewers
 * [Justin Lu](https://openjdk.org/census#jlu) (@justin-curtis-lu - Committer) ⚠️ Review applies to [f4a680c1](https://git.openjdk.org/jdk/pull/19145/files/f4a680c1df88ab057b750b393b35759545bd626f)
 * [Steven Loomis](https://openjdk.org/census#srl) (@srl295 - Committer)
 * [Joe Wang](https://openjdk.org/census#joehw) (@JoeWang-Java - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/19145/head:pull/19145` \
`$ git checkout pull/19145`

Update a local copy of the PR: \
`$ git checkout pull/19145` \
`$ git pull https://git.openjdk.org/jdk.git pull/19145/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 19145`

View PR using the GUI difftool: \
`$ git pr show -t 19145`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/19145.diff">https://git.openjdk.org/jdk/pull/19145.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/19145#issuecomment-2101368209)

### CODE REVIEW

**Review by justin-curtis-lu — APPROVED** [review]
LGTM. Wording in the beginning of the class spec makes it clear that locale sensitive factory methods are dependent on the LSP. By now adding wording that the CLDR data changes, it should hopefully help to reduce the expectation from users of consistent data between releases for these locale sensitive methods.

File: src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java [hunk 1]
```diff
@@ -152,6 +152,9 @@
  * supported by the Java runtime environment. The following table lists the
  * version of CLDR used in each JDK release. Unless otherwise specified, all
  * update releases in a given JDK release family use the same CLDR version.
+ * Note that the CLDR locale data are subject to change. Users should not assume
+ * that the locale data remain the same across CLDR versions. Otherwise, unexpected
+ * incompatible behaviors may occur, such as an exception on parsing a date.
```
Comment (JoeWang-Java) [inline src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java:157]: For those wondering what has changed from one version to another or whether a particular change might have an impact after reading this note, I wonder if some kind of notes or links might be helpful. A column to the table below to add some notes? I'm sure they can always look at the CLDR release notes. Just wonder if there's anything specific worth noting or a link.

Comment (srl295) [reply src/java.base/share/classes/java/util/spi/LocaleServiceProvider.java:157]: right.. there's never "no change".. and any change could potentially break users if they are expecting identical behavior.

**Review by srl295 — APPROVED** [review]

**Review by JoeWang-Java — APPROVED** [review]
That sounds good.

**Pull request closed without merging**